src/Simplify.elm

```diff
@@ -3605,6 +3719,228 @@ listIndexedMapChecks { lookupTable, fnRange, firstArg } =
+                            numberBinaryOperationChecks operation =
+                                [ Rule.errorWithFix
+                                    { message = "Use List." ++ operation.list ++ " instead"
+                                    , details =
+                                        [ "Using List." ++ foldOperationName ++ " (" ++ operation.two ++ ") " ++ String.fromInt operation.identity ++ " is the same as using List." ++ operation.list ++ "." ]
+                                    }
+                                    checkInfo.fnRange
+                                    (if getUncomputedNumberValue initialArgument == Just (Basics.toFloat operation.identity) then
+                                        [ Fix.replaceRangeBy
+                                            { start = checkInfo.parentRange.start
+                                            , end = (Node.range initialArgument).end
+                                            }
+                                            ("List." ++ operation.list)
+                                        ]
+
+                                     else
+                                        case checkInfo.thirdArg of
+                                            Nothing ->
+                                                -- fold op initial --> (List.op >> op initial)
+                                                -- TODO Can we call this a simplification?
+                                                --
+                                                -- why parens? edge cases with operator precedence
+                                                -- List.foldl (+) 0 << List.map .age
+                                                -- --> List.sum >> (+) 0 << List.map .age -- error!
+                                                --
+                                                [ Fix.insertAt checkInfo.parentRange.end ")"
+                                                , Fix.removeRange
+                                                    { start = checkInfo.parentRange.start
+                                                    , end = (Node.range initialArgument).start
+                                                    }
+                                                , Fix.insertAt (Node.range initialArgument).start
+                                                    (" >> (" ++ operation.two ++ ") ")
+                                                , Fix.insertAt checkInfo.parentRange.start
+                                                    ("(List." ++ operation.list)
+                                                ]
+
+                                            Just _ ->
+                                                if checkInfo.usingRightPizza then
+                                                    -- list |> fold op initial --> (list |> List.op) + initial
```

Comment:
Does this not create problems like if you have
```elm
a
  |> f
  |> List.foldl (+) initial
  |> g
-->
(a
  |> f
  |> List.sum
) + initial
  |> g
```
I'm always worried that this doesn't work or not. Like is this `(...) + (initial |> g)` or is it `((...) + initial) |> g`? When I write this code manually, I always add parentheses because it scares me that it's not going to do what I hoped.

Reply:
Good catch